MarkupViewer.py

```diff
@@ -850,15 +847,45 @@ def __init__(self, parent=None):
             for b in (spacer, self.download, self.go_to_gh, self.cancel):
                 tb.addWidget(b)
 
-            self.addToolBar(0x8, tb)
+            self.addToolBar(0x8, tb) 
+
+def error(msg):
+    print(msg)
+    exit()
+
+class GeoAction(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs:
+            error("Too many arguments for --geometry")
+
+        super().__init__(option_strings, dest, **kwargs)
+
+    def __call__(self, parser, namespace, value, option_string=None):
+        try:
+            values = value.split("x")
+            setattr(namespace, self.dest, (int(values[0]), int(values[1])))
+        except:
+            error("Invalid option " + value + " for --geometry!")
```

Comment:
Might as well show an example of a valid option.

Reply:
I have worked out the other issues but don't understand what you exactly meant here. And also thank you for feedback.